### PR TITLE
All experiments are private

### PR DESCRIPTION
**Pull request by marcel-vesely-kinit**
All the endpoints regarding the individual experiments and their corresponding experiment runs are tied to their original authors and only said authors can access their own experiments / experiment runs.

Additionally changes made:
* Refactoring of the authenticated endpoints
* Renaming and slight logic change to the experiment and experiment template filters, i.e. `only_mine`, `include_pending`, `only_finalized`
* query operator is now removed from endpoints and only used internally

### CODE REVIEW

File: backend/app/routers/experiments.py
Comment (mtkinit) [inline backend/app/routers/experiments.py:52]: It would be good to either have a method like "user_experiments" or to have a test. Maybe in the next run?

File: backend/app/routers/experiments.py
Comment (mtkinit) [inline backend/app/routers/experiments.py:73]: It would be beneficial to have it more explicit. There are multiple options:
1.
assert await check_experiment_access(id, user) == true, ...

2.
await raise_exception_if_no_experiment_access(id, user) # ugly name, but you get the point)

File: backend/app/routers/experiments.py
Comment (mtkinit) [inline backend/app/routers/experiments.py:164]: This is also required=True

File: backend/app/routers/experiments.py
Comment (mtkinit) [inline backend/app/routers/experiments.py:191]: required=True

File: backend/app/routers/experiments.py
Comment (mtkinit) [inline backend/app/routers/experiments.py:208]: required=True

File: backend/app/routers/experiments.py
Comment (mtkinit) [inline backend/app/routers/experiments.py:226]: required=True

File: backend/app/routers/experiments.py
Comment (mtkinit) [inline backend/app/routers/experiments.py:267]: required=True

**Review by mtkinit — CHANGES_REQUESTED** [review]
Please, chech the comments. The major requirement is to change "required=False" to "required=True" when accessing ExperimentRuns.

File: frontend/src/app/components/experiments/experiment-template-lists/all-experiment-template-list.component.ts
Comment (andrejridzik) [reply frontend/src/app/components/experiments/experiment-template-lists/all-experiment-template-list.component.ts:26]: I have specified always only those that differ from the default values. Additionally, we want to show pending templates in My Templates, but not in the All tab (that would be messy).

Comment (andrejridzik) [reply frontend/src/app/components/experiments/experiment-template-lists/all-experiment-template-list.component.ts:26]: Additionally_2, only non-pending templates can move to finalized state. Thus `only_finalized: true` already implies `include_pending: false`

File: backend/app/routers/experiments.py
Comment (andrejridzik) [reply backend/app/routers/experiments.py:52]: You mean something like `Experiment.find_by_user(user)`? I would like that, but we would also need to create a proper User class as it is only dict for now. If you agree I can add it in a separate PR.

File: backend/app/routers/experiments.py
Comment (andrejridzik) [reply backend/app/routers/experiments.py:73]: I am also not a big fan of calling function in this way, but I could not find the best way to to it here. I assume this will be changed once we have a proper access management in place. But from your options I would vote for 2 as it is the most explicit I guess.. I will think about it a bit

File: backend/app/routers/experiments.py
Comment (andrejridzik) [reply backend/app/routers/experiments.py:164]: Are you sure you want all of those? The idea was that this is ready for public experiments. We are checking the access by `check_experiment_access()` anyway.

Comment (mtkinit) [reply backend/app/routers/experiments.py:163]: I'm wondering if it's not better to lock it now (when there are no public experiments) and then unlock it once public experiments are introduced?

File: frontend/src/app/components/experiments/experiment-template-lists/all-experiment-template-list.component.ts
Comment (andrejridzik) [reply frontend/src/app/components/experiments/experiment-template-lists/all-experiment-template-list.component.ts:26]: Just to be completely clear here, the lifecycle of template goes like
* `PENDING` - waiting for approval
* `IN_PROGRESS` - image being built
* `FINISHED` - ready to be used in experiments

**Review by andrejridzik — APPROVED** [review]

**Review by mtkinit — APPROVED** [review]

**Pull request merged**
